File: shoring/blackjackEngine/src/test/java/com/panopset/tests/blackjackEngine/Soft18vThit8thenBtest.java
package com.panopset.tests.blackjackEngine;

import static com.panopset.blackjackEngine.CommandDefinitions.CMD_DEAL;
import static com.panopset.blackjackEngine.CommandDefinitions.CMD_HIT;
import org.junit.jupiter.api.Test;

import com.panopset.blackjackEngine.BlackjackConfigDefault;
import com.panopset.blackjackEngine.BlackjackGameEngine;

public class Soft18vThit8thenBtest extends AbstractBlackjackTest {
  
  @Test
  void test() {
    BlackjackGameEngine bge = new BlackjackGameEngine(new BlackjackConfigDefault());
    verifyRecommendedActions(bge, new String[] {CMD_DEAL, CMD_HIT, CMD_HIT}, soft18vThit8thenBtest());
  }
}

File: code/shoring/blackjackEngine/src/test/kotlin/com/panopset/tests/blackjackEngine/Soft18vThit8thenBtest.kt
package com.panopset.tests.blackjackEngine

import com.panopset.blackjackEngine.*
import org.junit.jupiter.api.Test

class Soft18vThit8thenBtest {
    @Test
    fun test() {
        val bge = BlackjackGameEngine(BlackjackConfigDefault())
        verifyRecommendedActions(bge, arrayOf(CMD_DEAL, CMD_HIT, CMD_HIT), soft18vThit8thenBtest())
    }
}
